### PR TITLE
feat(formula): add the Maxifs function

### PR DESCRIPTION
**Pull request by karlsbeard**
* feat: update the i18n file
* feat: add the formula description
* feat: add the formula algorithm
* feat: add the unit test
* feat: complete the function test
* fix: change sumRange to maxRange

### CODE REVIEW

File: packages/sheets-formula/src/services/function-list/statistical.ts [hunk 1]
```diff
@@ -1349,19 +1349,40 @@ export const FUNCTION_LIST_STATISTICAL: IFunctionInfo[] = [
         abstract: 'formula.functionList.MAXIFS.abstract',
         functionParameter: [
             {
-                name: 'formula.functionList.MAXIFS.functionParameter.number1.name',
-                detail: 'formula.functionList.MAXIFS.functionParameter.number1.detail',
+                name: 'formula.functionList.MAXIFS.functionParameter.sumRange.name',
```
Comment (Dushusir) [inline packages/sheets-formula/src/services/function-list/statistical.ts:1352]: maxRange

**Review by Dushusir — APPROVED** [review]

**Comment by Dushusir** [comment]
very good code !

File: packages/engine-formula/src/functions/statistical/maxifs/__tests__/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { describe, expect, it } from 'vitest';
+import { FUNCTION_NAMES_STATISTICAL } from '../../function-names';
+import { Maxifs } from '..';
```
Comment (wzhudev) [inline packages/engine-formula/src/functions/statistical/maxifs/__tests__/index.spec.ts:19]: This import path should be fixed.

File: packages/engine-formula/src/functions/statistical/maxifs/__tests__/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import { ArrayValueObject, transformToValue } from '../../../../engine/value-object/array-value-object';
+import { NumberValueObject, StringValueObject } from '../../../../engine/value-object/primitive-object';
+
+describe('Test maxifs function', () => {
+    const textFunction = new Maxifs(FUNCTION_NAMES_STATISTICAL.MAXIFS);
+
+    describe('Maxifs', () => {
+        it('Range and criteria', async () => {
+            const maxRange = ArrayValueObject.create(/*ts*/ `{
```
Comment (wzhudev) [inline packages/engine-formula/src/functions/statistical/maxifs/__tests__/index.spec.ts:28]: What does these `/*ts*/` do here?

**Review by wzhudev — CHANGES_REQUESTED** [review]
Overall LGTM. A minor path issue need to be fixed.

**Review by wzhudev — APPROVED** [review]
LGTM. Well done.

**Pull request merged**